### PR TITLE
docs: tweak windows functions examples to include order by

### PR DESCRIPTION
**Pull request by kay-kim**
Add order by clause and a small fix to the lag example suggestion to use RIGHT join instead of LEFT join.
- Also, small tweak to change `name` to `city` in the table.

### CODE REVIEW

File: doc/user/content/transform-data/patterns/top-k.md [hunk 1]
```diff
@@ -21,21 +21,22 @@ databases, you might use window functions. In Materialize, we recommend using a
...
+ORDER BY key_col, order_col
 ```
 
 For example, suppose you have a relation containing the population of various
 U.S. cities.
 
 ```mzsql
 CREATE TABLE cities (
-    name text NOT NULL,
+    city text NOT NULL,
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/top-k.md:39]: Just changed `name` to `city`.

File: doc/user/content/transform-data/patterns/top-k.md [hunk 2]
```diff
@@ -57,45 +58,53 @@ INSERT INTO cities VALUES
...
+
 ```nofmt
 Explained Query:
-  Project (#1, #0)
-    TopK group_by=[#1] order_by=[#2 desc nulls_first] limit=3
-      ReadStorage materialize.public.cities
+  Finish order_by=[#0{state} asc nulls_last, #2{pop} desc nulls_first] output=[#0..=#2]
+    TopK group_by=[#0{state}] order_by=[#2{pop} desc nulls_first] limit=3 // { arity: 3 }
+      Project (#1{city}, #0{state}, #2{pop}) // { arity: 3 }
+        ReadStorage materialize.public.cities // { arity: 3 }
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/top-k.md:99]: I see the TopK group_by in the explain plan, so I'm thinking that's good.

W/o the order by, the explain is:
```
Explained Query:
  Project (#1{city}, #0{state}) // { arity: 2 }
    TopK group_by=[#1{state}] order_by=[#2{pop} desc nulls_first] limit=3 // { arity: 3 }
      ReadStorage materialize.quickstart2.cities // { arity: 3 }
```
So ... I'm hoping the diff doesn't indicate a significant perf hit.

File: doc/user/content/transform-data/patterns/window-functions.md [hunk 3]
```diff
@@ -103,16 +107,27 @@ INSERT INTO measurements VALUES
...
+SELECT m2.time, m2.value - m1.value as diff_w_prev
 FROM measurements m1, measurements m2
-WHERE m2.time = m1.time + INTERVAL '1' MINUTE;
+WHERE m2.time = m1.time + INTERVAL '1' MINUTE
+ORDER BY m2.time;
 ```
 
-Note that these queries differ in whether they include the first timestamp (with a `NULL` difference). Using a `LEFT JOIN` would make the outputs match exactly.
+Note that these queries differ in whether they include the first timestamp (with
+a `NULL` difference).  A `RIGHT JOIN` would make the outputs match exactly.
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/window-functions.md:125]: I changed the Left Join to a Right Join as I think the Left join was a typo?

Comment (morsapaes) [reply doc/user/content/transform-data/patterns/window-functions.md:125]: That's right. It'd be a `LEFT JOIN` only if you changed the order of the tables in the `FROM` clause, which...doesn't make sense, anyways!

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/transform-data/patterns/window-functions.md (hunk 3 above)
Comment (morsapaes) [reply doc/user/content/transform-data/patterns/window-functions.md:125]: Similarly, I'd write the previous query as:

```sql
SELECT m1.time, m1.value - m2.value
FROM measurements m1, measurements m2
WHERE m1.time = m2.time + INTERVAL '1' MINUTE
ORDER BY m1.time;
```

Comment (kay-kim) [reply doc/user/content/transform-data/patterns/window-functions.md:126]: Oh, great minds think alike.  To keep the changes small, I didn't change it ... but I'm with you -- it does look weird.  I'll patch it.

**Pull request merged**